lib/livebook/session.ex

```diff
@@ -2661,20 +2661,15 @@ defmodule Livebook.Session do
         {:error, "no file exists at path #{inspect(file.path)}"}
       end
     else
+      type = Livebook.FileSystems.type(file.file_system)
+      dumped_data = FileSystem.dump(file.file_system)
+      "/" <> key = file.path
+
       spec =
-        case file.file_system do
-          %FileSystem.S3{} = file_system ->
-            "/" <> key = file.path
-
-            %{
-              type: :s3,
-              bucket_url: file_system.bucket_url,
-              region: file_system.region,
-              access_key_id: file_system.access_key_id,
-              secret_access_key: file_system.secret_access_key,
-              key: key
-            }
-        end
+        Map.merge(dumped_data, %{
+          type: String.to_existing_atom(type),
+          key: key
+        })
```

Comment:
Sorry for the nitpick but I think we should revert this change. This is about exporting file system data to Kino, and we may need to pick a different format than the one we persist internally (i.e. the one we use on dump). Plus `file.path` assumes it is a FileSystem.S3 anyway. :) But yeah, I think we should keep these decoupled right now, but I think it is good you chose to find ways to refactor the existing code. :+1: